### PR TITLE
expose inverse_mass_matrix to hmc interface

### PR DESCRIPTION
**Pull request by fehiepsi**
Partly addresses #438.

### CODE REVIEW

File: numpyro/infer/mcmc.py [hunk 1]
```diff
@@ -187,6 +188,8 @@ def init_kernel(init_params,
         :param float step_size: Determines the size of a single step taken by the
             verlet integrator while computing the trajectory using Hamiltonian
             dynamics. If not specified, it will be set to 1.
+        :param numpy.ndarray inverse_mass_matrix: Inverse of the mass matrix.
```
Comment (neerajprad) [inline numpyro/infer/mcmc.py]: nit: How about rewording this to - "Initial value for inverse mass matrix. This may be adapted during warmup if `adapt_mass_matrix = True`. If no value is specified, then it is initialized to the identity matrix."?

Comment (fehiepsi) [reply numpyro/infer/mcmc.py]: Thanks, Neeraj!
